scripts/2.TCGA-MLexample-covariates.py

```diff
@@ -281,66 +300,52 @@ def get_threshold_metrics(y_true, y_pred):
     for model, pipeline in final_pipelines.items()
 }
 
+coef_df = pd.concat([
+    get_model_coefficients(classifier, model, covariates.columns)
+    for model, classifier in final_classifiers.items()
+])
 
-# In[22]:
 
-def get_coefficients(classifier, X_mat):
-    coef_df = pd.DataFrame.from_items([
-        ('feature', X_mat.columns),
-        ('weight', classifier.coef_[0]),
-    ])
+# In[21]:
 
-    coef_df['abs'] = coef_df['weight'].abs()
-    coef_df = coef_df.sort_values('abs', ascending=False)
-    
-    return coef_df
+def zero(w): return '{:.1%}'.format(np.mean(w == 0))
+def negative(w): return '{:,}'.format(np.sum(w < 0))
+def positive(w): return '{:,}'.format(np.sum(w > 0))
 
-coef_df_dict = {
-    model: get_coefficients(classifier, X_partitions[model]['train'])
-    for model, classifier in final_classifiers.items()
-}
+coef_df.groupby('feature_set')['weight'].agg([zero, negative, positive])
 
 
-# In[23]:
+# In[22]:
 
 model = 'full'
-
-print('{:.1%} zero coefficients; {:,} negative and {:,} positive coefficients'.format(
-    (coef_df_dict[model].weight == 0).mean(),
-    (coef_df_dict[model].weight < 0).sum(),
-    (coef_df_dict[model].weight > 0).sum()
-))
-coef_df_dict[model].head(10)
+model_coef_df = coef_df[coef_df['feature_set'] == model]
+model_coef_df.head(10)
 
```

Comment:
I agree with this sentiment, and am curious to find out if this is actually important to a biologist.